lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2522,6 +2522,18 @@ namespace {
           VarDecl *pointeeProperty =
               synthesizer.makeDereferencedPointeeProperty(
                   getterAndSetter.first, getterAndSetter.second);
+
+          // Import the attributes from dereference operator clang decl to
+          // pointee property swift decl.
+          FuncDecl *setterImpl = getterAndSetter.second;
+          FuncDecl *getterImpl = getterAndSetter.first ? getterAndSetter.first
+                                                       : getterAndSetter.second;
+          if (auto clangDeclForOperatorStar =
+                  dyn_cast_or_null<clang::NamedDecl>(
```

Comment:
Can `getterImpl->getClangDecl()` ever be null? Looks like it would always be non-null, so I think we should just unconditionally `cast<clang::NamedDecl>` and have an assertion if the getter's Clang decl ended up being null, since that would indicate a problem in our reasoning about this code path.

Reply:
Yes, `getterImpl->getClangDecl()` was null for some lit test cases like these:

// ***------- TEST 1
// Swift:
```
let derivedConstIterWithUD = DerivedFromConstIteratorPrivatelyWithUsingDecl()
let _ = derivedConstIterWithUD.pointee
```
// C++:
```
struct ConstIterator {
private:
  int value = 234;
public:
    int &operator*() { return value; }
};
struct DerivedFromConstIteratorPrivatelyWithUsingDecl : private ConstIterator {
  using ConstIterator::operator*;
};

```
// ***-------  TEST 2
// Swift:
```
let stars = DerivedFromConstIteratorPrivatelyWithUsingDecl()
let res = stars.pointee
```
// C++:
```
struct ConstIterator {
private:
  int value = 234;
public:
  const int &operator*() const { return value; }
};
struct DerivedFromConstIteratorPrivatelyWithUsingDecl : private ConstIterator {
  using ConstIterator::operator*;
};
```

// ***-------  TEST 3
// Swift:
```
let stars = DerivedFromAmbiguousOperatorStarPrivatelyWithUsingDecl()
let res = stars.pointee
```
// C++:
```
struct AmbiguousOperatorStar {
private:
  int value = 567;
public:
  int &operator*() { return value; }
  const int &operator*() const { return value; }
};
struct DerivedFromAmbiguousOperatorStarPrivatelyWithUsingDecl
    : private AmbiguousOperatorStar {
  using AmbiguousOperatorStar::operator*;
};
```

So maybe the problem is when calling pointee property C++ types which inherits operator* from their parent/base class like using `AmbiguousOperatorStar::operator*` and `using ConstIterator::operator*`
@egorzhdan do you have thoughts about why `getterImpl->getClangDecl()` is null for these cases? I was assuming that `__myoperatorstar` function should always have a `clangDecl` from which it is mapped.

Reply:
Oh right, that makes sense. C++ members that were instantiated from using-shadow-decls have no corresponding Clang decl. This is because we synthesize a Swift decl that dispatches the call to the underlying Clang decl.
(Ideally we would synthesize a decl on the Clang side, not on the Swift side, to help us avoid this kind of issues)
Could you please add a comment mentioning the inheritance quirk?